app/components/settings/categories/GeneralSettings.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+import React, { Component, PropTypes } from 'react';
+import { observer, PropTypes as MobxPropTypes } from 'mobx-react';
+import classNames from 'classnames';
+import Dropdown from 'react-toolbox/lib/dropdown/Dropdown';
+import { defineMessages, intlShape } from 'react-intl';
+import ReactToolboxMobxForm from '../../../lib/ReactToolboxMobxForm';
+import LocalizableError from '../../../i18n/LocalizableError';
+import styles from './GeneralSettings.scss';
+
+const messages = defineMessages({
+  languageSelectLabel: {
+    id: 'settings.general.languageSelect.label',
+    defaultMessage: '!!!Language',
+    description: 'Label for the language select.'
+  },
+});
+
+@observer
+export default class GeneralSettings extends Component {
+
+  static propTypes = {
+    languages: MobxPropTypes.arrayOrObservableArrayOf(PropTypes.shape({
+      value: PropTypes.string.isRequired,
+      label: PropTypes.object.isRequired,
```

Comment:
It might be a good idea to introduce a prop type for the expected translations schema here.
You can put it into `app/propTypes.js`:

```js 
export const translationMessageParams = PropTypes.shape({
  id: PropTypes.string.isRequired,
  defaultMessage: PropTypes.string.isRequired,
  description: PropTypes.string,
});
```